### ISSUE TITLE
create setter by field path

### ISSUE THREAD

**Issue by papagalu** [GoogleContainerTools/kpt#617]
i don't understand how to set a filed if i have 2 resources of the same type, eg:
i have 2 deployments, and i want to create 2 setters for each one of them, it's not clear to me how to create a setter that is bound to a specific resource, from what i understand, we can only set types?

Any help is much appreciated! Thank you!

**Comment by phanimarupaka** [comment]
Setters can be unique for filed and fieldValue combination. Can you walk through an example and expected behavior so that I could help.

#### scope creating setter references to a specified field
`kpt cfg create-setter DIR/ replicas 3 --field "replicas"`

**Comment by rumstead** [comment]
@papagalu not to steal your thread here but I think we share the problem.
@phanimarupaka is this a use case supported of using `kpt` and `kustomize`. 
```
.
├── README.md
├── base
│   ├── deployment.yaml
│   └── kustomization.yaml
├── prod
│   ├── kustomization.yaml
│   └── patch.yaml
├── dev
│   ├── kustomization.yaml
│   ├── patch.yaml
│   └── patch_init.yaml
└── qa
    ├── kustomization.yaml
    └── patch.yaml
```
I want to provide setters or substitutions on the `kustomization.yaml` or even the `patch.yaml` for a given environment. Say a developer wants to change their image version or maybe we run different memory limits across the environments. Being able to use `kpt` setter's is appealing. However, since the `kpt` pattern is matched by multiple files it updates them all disregarding the directory.

TL;DR: Is it possible to scope the setter to a specific resource + field?

**Comment by rumstead** [comment]
https://github.com/kubernetes-sigs/kustomize/blob/master/cmd/config/docs/commands/set.md looks to provide what I am looking for.

**Comment by phanimarupaka** [comment]
This will be implemented as part of search and replace command which is being tracked here. https://github.com/GoogleContainerTools/kpt/issues/961

Marking this as a duplicate.
